Add ThicknessMap extension

fatihtalu opened the pull request:
# New extension

- [x] Extension has a reasonable name (not too general, not too narrow, suggests what the extension is for)
- [x] Repository name is Slicer+ExtensionName
- [x] Repository is associated with `3d-slicer-extension` GitHub topic so that it is listed [here](https://github.com/topics/3d-slicer-extension). To edit topics, click the settings icon in the right side of "About" section header and enter `3d-slicer-extension` in "Topics" and click "Save changes". To learn more about topics, read https://help.github.com/en/articles/about-topics
- [x] Extension description summarizes in 1-2 sentences what the extension is usable (should be understandable for non-experts)
- [x] Any known related patents must be mentioned in the extension description.
- [x] If source code license is more restrictive for users than BSD, MIT, Apache, or 3D Slicer license then the name of the used license must be mentioned in the extension description.
- [x] Extension URL and revision (scmurl, scmrevision) is correct, consider using a branch name (main, master, release, ...) instead of a specific git has to avoid re-submitting pull request whenever the extension is updated
- [x] Extension icon URL is correct (do not use the icon's webpage but the raw data download URL that you get from the download button - it should look something like this: https://raw.githubusercontent.com/user/repo/main/SomeIcon.png)
- [x] Screenshot URLs (screenshoturls) are correct, contains at least one
- [x] Homepage URL points to valid webpage containing the following:
  - [x] Extension name
  - [x] Short description: 1-2 sentences, which summarizes what the extension is usable for
  - [x] At least one nice, informative image, that illustrates what the extension can do. It may be a screenshot.
  - [x] Description of contained modules: at one sentence for each module
  - [x] Tutorial: step-by-step description of at least the most typical use case, include a few screenshots, provide download links to sample input data set
  - [x] Publication: link to publication and/or to PubMed reference (if available)
  - [x] License: We suggest you use a permissive license that includes patent and contribution clauses.  This will help protect developers and ensure the code remains freely available.  We suggest you use the [Slicer License](https://github.com/Slicer/Slicer/blob/master/License.txt) or the [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0). Always mention in your README file the license you have chosen.  If you choose a different license, explain why to the extension maintainers. Depending on the license we may not be able to host your work. Read [here](https://opensource.guide/legal/#which-open-source-license-is-appropriate-for-my-project) to learn more about licenses.
  - [x] Content of submitted s4ext file is consistent with the top-level CMakeLists.txt file in the repository (description, URLs, dependencies, etc. are the same)

mfatihtalu committed:
Add files via upload

jcfr commented:
Thanks for your patience :pray: 

Before we can more forward with the integration, here few issues that you may want to address:
* https://github.com/fatihtalu/Slicer-ThicknessMap/issues/2
* https://github.com/fatihtalu/Slicer-ThicknessMap/issues/1
* https://github.com/fatihtalu/Slicer-ThicknessMap/issues/3
* https://github.com/fatihtalu/Slicer-ThicknessMap/issues/4
* https://github.com/fatihtalu/Slicer-ThicknessMap/issues/5
* https://github.com/fatihtalu/Slicer-ThicknessMap/issues/6

jcfr committed:
ENH: Convert ThicknessMap s4ext to json file

Also set `build_subdirectory` to `.` instead of `inner-build` (the former
is specific to SuperBuild extension).

lassoan commented:
We are cleaning out our backlog of open pull requests. We have introduced tiers for extensions, making it easier to add experimental extensions to the index. @fatihtalu if you want your extension to be added to the index then let us know and we can add it as a tier 1. If we don't hear from you then we will close the pull request (but you can reopen it anytime).

lassoan commented:
Closing due to no response from the submitter.
